app/javascript/components/student/Editor.tsx

```diff
@@ -11,12 +11,13 @@ import { FileEditor, FileEditorHandle } from './editor/FileEditor'
 import { fetchJSON } from '../../utils/fetch-json'
 import { typecheck } from '../../utils/typecheck'
 import { Iteration } from '../track/IterationSummary'
+import { useIsMounted } from 'use-is-mounted'
 
 export type Submission = {
   testsStatus: SubmissionTestsStatus
   uuid: string
   links: SubmissionLinks
-  testRun: TestRun
+  testRun?: TestRun
 }
 
 type APIError = {
```

Comment:
I recommend using the error type given above. There will be times where compound errors will be a thing:
```json
{
  "errors": [
    { "message": "Something went wrong" },
    { "message": "Second error", "param": "value", "code": "whatever" }
  ]
}
```

Nesting inside `errors` allows _request_ metadata to be present at the top-level.